tests/components/azure_event_hub/test_init.py

```diff
@@ -90,16 +76,94 @@ async def test_full_config(hass):
     }
-    config[azure_event_hub.DOMAIN].update(MIN_CONFIG)
     assert await async_setup_component(hass, azure_event_hub.DOMAIN, config)
 
 
+async def test_setup(hass, mock_hub):
+    """Test the async_setup function."""
+    entry = MockConfigEntry(
+        domain=azure_event_hub.DOMAIN,
+        data=SAS_CONFIG_FULL,
+        title="test-instance",
+        options=BASIC_OPTIONS,
+    )
+    entry.add_to_hass(hass)
+    assert await azure_event_hub.async_setup_entry(hass, entry)
+    assert hass.data[azure_event_hub.DOMAIN]["hub"] is not None
+    assert isinstance(hass.data[azure_event_hub.DOMAIN]["hub"], AzureEventHub)
+
+
+async def test_unload_entry(hass, mock_hub):
+    """Test being able to unload an entry."""
+    entry = MockConfigEntry(
+        domain=azure_event_hub.DOMAIN,
+        data=SAS_CONFIG_FULL,
+        title="test-instance",
+        options=BASIC_OPTIONS,
+    )
+    entry.add_to_hass(hass)
+    assert await azure_event_hub.async_setup_entry(hass, entry)
+    assert hass.data[azure_event_hub.DOMAIN].get("hub") is not None
+    assert await azure_event_hub.async_unload_entry(hass, entry)
+    assert hass.data[azure_event_hub.DOMAIN].get("hub") is None
+
+
+async def test_failed_test_connection(hass):
+    """Test being able to unload an entry."""
+    entry = MockConfigEntry(
+        domain=azure_event_hub.DOMAIN,
+        data=SAS_CONFIG_FULL,
+        title="test-instance",
+        options=BASIC_OPTIONS,
+    )
+    entry.add_to_hass(hass)
+    with patch(
+        f"{PRODUCER_PATH}.get_eventhub_properties",
+        side_effect=EventHubError("test"),
+    ):
+        try:
+            await azure_event_hub.async_setup_entry(hass, entry)
+        except azure_event_hub.ConfigEntryNotReady:
+            pass
+        assert hass.data[azure_event_hub.DOMAIN].get("hub") is None
+
+
+async def test_update_listener(hass, mock_hub):
+    """Test being able to update options."""
+    entry = MockConfigEntry(
+        domain=azure_event_hub.DOMAIN,
+        data=SAS_CONFIG_FULL,
+        title="test-instance",
+        options=BASIC_OPTIONS,
+    )
+    entry.add_to_hass(hass)
+    assert await azure_event_hub.async_setup_entry(hass, entry)
+    entry.options = UPDATE_OPTIONS
+    await azure_event_hub.async_update_listener(hass, entry)
+    assert (
+        hass.data[azure_event_hub.DOMAIN]["hub"].send_interval
+        == UPDATE_OPTIONS[CONF_SEND_INTERVAL]
+    )
+
+
 async def _setup(hass, mock_call_later, filter_config):
     """Shared set up for filtering tests."""
-    config = {azure_event_hub.DOMAIN: {"filter": filter_config}}
-    config[azure_event_hub.DOMAIN].update(MIN_CONFIG)
-
-    assert await async_setup_component(hass, azure_event_hub.DOMAIN, config)
+    entry = MockConfigEntry(
+        domain=DOMAIN,
+        data=SAS_CONFIG_FULL,
+        title="test-instance",
+        options=BASIC_OPTIONS,
+    )
+    hass.data[DOMAIN] = {DATA_FILTER: FILTER_SCHEMA(filter_config)}
+    hub = AzureEventHub(
```

Comment:
Don't access any details of the integration. Just set up the integration and assert core state and library mock calls.